src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import duke.models.Task;
+
+
+
+/**
+ * Contains the task list and has operations to add/delete tasks in the list
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Initializes an empty duke.TaskList
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Initializes a taskList that contains existing tasks that have been saved
+     * @param taskList List of existing tasks to be loaded
+     */
+    public TaskList(List<Task> taskList) {
+        this.tasks = taskList;
+    }
+
+    public void addTask(Task t) {
+        tasks.add(t);
+    }
+
+    public void deleteTask(int index) {
+        this.tasks.remove(index - 1);
```

Comment:
Perhaps the `this` keyword should be used only when a field is shadowed by a method or constructor parameter?
```suggestion
        tasks.remove(index - 1);
```